cmd/otel-allocator/prehook/relabel.go

```diff
@@ -80,11 +83,30 @@ func (tf *RelabelConfigTargetFilter) Apply(targets map[string]*target.Item) map[
 func (tf *RelabelConfigTargetFilter) SetConfig(cfgs map[string][]*relabel.Config) {
 	relabelCfgCopy := make(map[string][]*relabel.Config)
 	for key, val := range cfgs {
-		relabelCfgCopy[key] = val
+		relabelCfgCopy[key] = tf.replaceRelabelConfig(val)
 	}
+
 	tf.relabelCfg = relabelCfgCopy
 }
 
+func (tf *RelabelConfigTargetFilter) replaceRelabelConfig(cfg []*relabel.Config) []*relabel.Config {
+	outCfg := make([]*relabel.Config, len(cfg))
+	out, err := yaml2.Marshal(cfg)
+	if err != nil {
+		tf.log.V(2).Info("Error Marshaling", "error", err)
+		return cfg
+	}
+
+	byteArr := replaceShard([]byte(out))
+	err = yaml2.Unmarshal(byteArr, &outCfg)
+	if err != nil {
+		tf.log.Info("Error Unmarshalling", "error", err)
+		return cfg
+	}
+
+	return outCfg
+}
+
```

Comment:
rather than marshaling and then unmarshaling, it's probably more efficient to just iterate through the relabel config list and do the replacement.

Reply:
Thanks for the review! Changed according to this suggestion.